polyglot.js

```diff
@@ -362,18 +380,86 @@ class PolyGlot {
+                        if (!lang) continue;
+                        texts.push(span.textContent)
+                        span.textContent = this.generateRune(span.textContent,game.settings.get('polyglot','salt') === true ? journalSheet._id : lang)
+                        span.style = this.tongues[lang] === undefined ? 'font:' + this.alphabets[this.tongues._default] : 'font:' + this.alphabets[this.tongues[lang]]
+                    }
+                }
+                else {
+                    const spans = journalSheet.element.find("span.polyglot-journal");
+                    var i = 0;
+                    for (let span of spans.toArray()) {
+                        const lang = span.dataset.language;
+                        if (!lang) continue;
+                        span.textContent = texts[i]
+                        span.style = ''
+                        i++;
+                    }
+                }
+            });
+            html.closest('.app').find('.polyglot-button').remove();
+            let titleElement = html.closest('.app').find('.window-title');
+            openBtn.insertAfter(titleElement);
+            return;
+        }
         const spans = journalSheet.element.find("span.polyglot-journal");
         for (let span of spans.toArray()) {
             const lang = span.dataset.language;
             if (!lang) continue;
-            if (!this.known_languages.has(lang))
-                span.textContent = span.textContent.replace(this.regexp_letters, this.randomRune)
+            if (!this.known_languages.has(lang)) {
+                span.title = "????"
+                span.textContent = this.generateRune(span.textContent,game.settings.get('polyglot','salt') === true ? journalSheet._id : lang)
+                span.style = this.tongues[lang] === undefined ? 'font:' + this.alphabets[this.tongues._default] : 'font:' + this.alphabets[this.tongues[lang]]
+            }
+        }
+    }
+    chatBubble (token, html, message, {emote}) {
+        message = game.messages._source[game.messages._source.length-1];
+        if (message.type == CONST.CHAT_MESSAGE_TYPES.IC) {
+            let lang = message.flags.polyglot.language || ""
+            if (lang != "") {
+                const unknown = !this.known_languages.has(lang);
+                message.polyglot_unknown = unknown;
+                if (game.user.isGM && !game.settings.get("polyglot", "runifyGM"))
+                    message.polyglot_unknown = false;
+                if (!message.polyglot_force && message.polyglot_unknown) {
+                    let content = html.find(".bubble-content")
+                    let new_content = this.generateRune(message.content,game.settings.get('polyglot','salt') === true ? message._id : lang)
+                    content.text(new_content)
+                    content[0].style = this.tongues[lang] === undefined ? 'font:' + this.alphabets[this.tongues._default] : 'font:' + this.alphabets[this.tongues[lang]]
```

Comment:
Thanks for using the new hook! 
Please consider the same comments as above : 
- use `scrambleString`
- use `const` instead of `let` for variables that you don't modify (both of them)
- use non destructive style assignment (`style.font = ...`)